### DISCUSSION TITLE
2020/04/20/migrating-from-jekyll-to-hugo-deploying-with-github-pages/

### DISCUSSION THREAD

**Discussion by bric3** [bric3/bric3.github.io#40] [Blog Comments]
# 2020/04/20/migrating-from-jekyll-to-hugo-deploying-with-github-pages/
Migrating From Jekyll to Hugo, deploying with Github Pages
https://blog.arkey.fr/2020/04/20/migrating-from-jekyll-to-hugo-deploying-with-github-pages/

**Comment by bric3** [comment]
Originally posted by @rucko24 on 2022-03-15T18:36:49Z

----

Very very nice my friend...

**Comment by bbokser** [comment]
Can you clarify the difference between the "publication" (master) and "default branch" (hugo-sources)? Is the "publication" branch kept empty?

**Reply by bric3** [reply]
Good point! At the time I performed the migration, 

* GitHub was building and deploying (the publication) from jekyll files on `master` branch
* GitHub could deploy (publish) HTMLs only from `master` branch

So when I chose Hugo, I wanted put Hugo on `master` and deploy on another branch, and it wasn't possible.

Since then GitHub added the option to choose branch.

<img width="1078" alt="Screenshot 2022-10-24 at 13 24 32" src="https://user-images.githubusercontent.com/803621/197515274-4816cab8-da12-463d-a124-137583bf68bb.png">
